### DISCUSSION TITLE
Positioning strapi

### DISCUSSION THREAD

**Discussion by jesperordrup** [strapi/strapi#6158] [Legacy Topics]
Hi strapi ppl

Fairly new to the strapi things. I've only created a few in house tools and one client. Mostly by using the Admin UI. I've yet to dive into extending and customizing using all things controllers/services etc. 

I've been reading up on strapi docs and trying to get a handle on how to do things. 

The postgres vs mongodb seems to be in two different states where you can expect some things to work on mongodb but not on the other. Ex. aggregations and maybe filters. Im still searching for the big picture of features supported by postgress, mongodb and rest, graphql (table=?)

How awesome is strapi actually? From start I assumed that strapi was for building small sites and blogs but I have become curious in how robust the api really is. How much data/complexity can the graphql endpoint handle it hold - is it "just" for cms/blog stuff or would you put your transaction data into the model?  I'm mainly interested in graphql.  

Asking another way - when not to use strapi :-)

The Admin UI is doing a great job making collections and relations work. So the questions asked for the grapql endpoint is also relevant here. Is the admin working well with a lot of relational data? And are there any ways to hide relations in admin ui but still works in graphql?

I would love to see a guide covering a minimal dev setup that covers 2 environments with dev and deploy.

Strapi is awesome - cant wait to dive more into it and

**Comment by sunnysonx** [comment]
I don't have a lot of experience with Strapi, just a few days, but from this point, I can say it is not just for small sites and blogs, for me, it handles more than 100k products for a few e-commerce websites with zero problems at all.

**Comment by Aurelsicoko** [comment]
Hey, I'll try my best to answer these questions :) 

1. Yes, you're right, you cannot do the exact same thing on Mongo and PostgreSQL. We have a plan to make the ability more consistent across databases. We will develop an abstract layer, like a query system that will translate the query to make it work whatever the database is. It will be similar to an ORM.

2. Strapi is designed to handle a large amount of data. We have to make choice and pick default settings but with a bit of customization you can build whatever you want on top of Strapi. For instance, we have projects running in production handling dozens of millions of requests per month. We also have projects which manage millions of relational data without any performance issues. However, the database has been optimized with indexes and the level of depth in a GraphQL query, for example, has to be optimized too.

3. Strapi is customizable, flexible even if one main use case is to be a CMS. So there is no use case where I won't recommend to use it. It can be a e-commerce or corporate website, a blog, a custom dashboard, an internal tool, etc.

4. In the administration panel, you can customize the layout (what you see) so yes you can pick the fields you want to contribute to, even if there are 10 relational fields hidden. Thanks to the new RBAC feature, you will also be able to define custom permissions per field according to the user's role.

**Reply by jesperordrup** [reply]
Thank you for the answers. Really happy to read this.
Are you considering making Graphql a "first class citizen" or will it stay as plugin, a second choice.

Thx
